IfcUnitaryControlElement Rail Import BIMQ 2021-07-21T07:49:23.6924211Z

jmirtsch opened the pull request:
Rail properties for IfcUnitaryControlElement

jmirtsch committed:
IfcUnitaryControlElement Rail Import BIMQ 2021-07-21T07:49:23.6924211Z

SergejMuhic commented on IFC4x3/Properties/r/ReferenceEnvironmentTemperature_2vgOshA4D0$ems5HSBwemr/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="ReferenceEnvironmentTemperature_2vgOshA4D0ems5HSBwemr" Name="ReferenceEnvironmentTemperature" UniqueId="b9a98dab-2843-40fe-8c36-15170bea8c35" PropertyType="p_boundedvalue" PrimaryDataType="IfcRatioMeasure" />
```
Secondary datatype should also be provided.

czapplitec replied:
It looks that all the existing bounded value properties only have Primary Data Type. The ifcDoc UI even greys the Select button for Secondary Data Type when it is a bounded value property. Shouldn't we consider the Secondary Data Type the same with Primary Data Type by default?

TLiebich replied:
an IfcPropertyBoundedValue has only one unit, and the measure type of all values (upper, lower, set point) shall be the same (enforced by where rules). So there is no secondary data type for bounded values - only IfcPropertyTableValue has two unit / measure type definitions.

NOTE: there seems to be an issue with the documentation of IfcPropertyTemplate.SecondaryMeasureType - P_BoundedValue should not be listed here, as it contradicts with the where rules at IfcPropertyBoundedValue. Should be added as a general issue on IFC4.3 (as it is not INFRA/RAIL specific).

czapplitec replied:
No need to change

SergejMuhic commented on IFC4x3/Properties/u/UndercontrolEquipmentsID_0uZsrOMH5AMBQLhPS3ukXv/Documentation.md:
```diff
@@ -0,0 +1 @@
+The identity of the equipment controlled by the power control system.
```
This is a bit vague as a description. How are these supposed to be formatted? GUIDs , IFC identifiers, STEP line IDs?

czapplitec replied:
This property is removed: https://github.com/bSI-InfraRoom/IFC-Specification/pull/191/commits/d82022836002a1eb860d4f69d1bdd8ad4615ef74

SergejMuhic commented on IFC4x3/Properties/u/UndercontrolEquipmentsID_0uZsrOMH5AMBQLhPS3ukXv/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="UndercontrolEquipmentsID_0uZsrOMH5AMBQLhPS3ukXv" Name="UndercontrolEquipmentsID" UniqueId="388f6d58-5911-4a58-b695-ad9703e2e879" PropertyType="p_listvalue" PrimaryDataType="IfcIdentifier" />
```
As `IfcUnitaryControlElement` is an assembly of control elements, why not just use an aggregation to determine this? Seems like overkill here.

czapplitec replied:
Same as above

SergejMuhic reviewed (changes requested):
These Psets seem to require some work.

SergejMuhic commented on IFC4x3/Properties/n/NumberOfInterfaces_3D0wHuI4P3guCzHoS3Y4Zf/Documentation.md:
```diff
@@ -0,0 +1 @@
+Indicate the types of interfaces and numbers of each type of them of the device.
```
```suggestion
Indicates the types of interfaces and numbers of each type of them of the device.
```

czapplitec committed:
Apply suggestions from code review

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

czapplitec committed:
Remove redundant properties

czapplitec committed:
Remove FunctionModuleDescription

czapplitec commented on IFC4x3/Properties/n/NominalPower_2pu16hiH1CqQv04BNgxAph/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="NominalPower_2pu16hiH1CqQv04BNgxAph" Name="NominalPower" UniqueId="b3e011ab-b110-4cd1-ae40-10b5eaecaceb" PrimaryDataType="IfcPowerMeasure" />
```
As discussed, this one is kept as to comply with the IEC standard.

czapplitec committed:
Removed UndercontrolEquipmentIDs

also removed one Pset. I think this pull request is ready to go if there are no more suggestions.

jmirtsch committed:
Explicit IfcLabel PrimaryDataType

SergejMuhic reviewed (commented):
Some comments are not yet resolved but the changes are fine.

czapplitec commented on IFC4x3/Properties/f/FunctionModuleDescription_14lu9Gskf3geBmw$v46Vl3/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
```
This property is removed, so this conversation can be resolved.

czapplitec commented on IFC4x3/Properties/n/NominalCurrent_1Sia_y4DvB595bUBLM1Dsb/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="NominalCurrent_1Sia_y4DvB595bUBLM1Dsb" Name="NominalCurrent" UniqueId="5cb24fbc-10de-4b14-9165-78b55604dda5" PrimaryDataType="IfcElectricCurrentMeasure" />
```
The property is reused and this one is removed.
https://github.com/bSI-InfraRoom/IFC-Specification/blob/950cbf95c312a345973973bc742af1ec316a695e/IFC4x3/Sections/Domain%20specific%20data%20schemas/Schemas/IfcRailDomain/PropertySets/Pset_UnitaryControlElementTypeControlPanel/DocPropertySet.xml#L4-L9

czapplitec commented on IFC4x3/Properties/r/RatedVoltage_0KkDpffYz8L86gzG31TCSi/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="RatedVoltage_0KkDpffYz8L86gzG31TCSi" Name="RatedVoltage" UniqueId="14b8dce9-a62f-4854-81aa-f500c174c72c" PrimaryDataType="IfcElectricVoltageMeasure" />
```
Reused an existing one. See here: https://github.com/bSI-InfraRoom/IFC-Specification/pull/191/commits/7a0be10635e115843e7b0b94ac32c1271461df43
